src/backup.ts

```diff
@@ -0,0 +1,516 @@
+   *
+   * @param {string} tableId The id of the table to create and restore to. This
+   *   table must not already exist.
+   * @param {CallOptions | RestoreTableCallback} [gaxOptionsOrCallback]
+   * @param {RestoreTableCallback} [callback] The callback function.
+   * @param {?error} callback.err An error returned while making this request.
+   * @param {Table} callback.table The newly created Table.
+   * @param {Operation} callback.operation An operation object that can be used
+   *     to check the status of the request.
+   * @param {object} callback.apiResponse The full API response.
+   * @return {void | Promise<RestoreTableResponse>}
+   */
+  restore(
+    tableId: string,
+    gaxOptionsOrCallback?: CallOptions | RestoreTableCallback,
+    cb?: RestoreTableCallback
+  ): void | Promise<RestoreTableResponse> {
+    const gaxOpts =
+      typeof gaxOptionsOrCallback === 'object' ? gaxOptionsOrCallback : {};
+    const callback =
+      typeof gaxOptionsOrCallback === 'function' ? gaxOptionsOrCallback : cb!;
+
+    this.bigtable.request<
+      LROperation<
+        google.bigtable.admin.v2.ITable,
+        google.bigtable.admin.v2.IRestoreTableMetadata
+      >
+    >(
+      {
+        client: 'BigtableTableAdminClient',
+        method: 'restoreTable',
+        reqOpts: {
+          parent: this.cluster.name,
+          tableId,
+          backup: this.name,
+        },
+        gaxOpts,
+      },
+      (err, ...args) => {
+        if (err) {
+          callback(err, undefined, ...args);
+          return;
+        }
+        callback(err, this.cluster.instance.table(tableId), ...args);
+      }
+    );
+  }
+
+  setMetadata(
+    metadata: ModifiableBackupFields,
+    gaxOptions?: CallOptions
+  ): Promise<SetMetadataResponse>;
+  setMetadata(
+    metadata: ModifiableBackupFields,
+    callback: SetMetadataCallback
+  ): void;
+  setMetadata(
+    metadata: ModifiableBackupFields,
+    gaxOptions: CallOptions,
+    callback: SetMetadataCallback
+  ): void;
+  /**
+   * Updates this pending or completed Cloud Bigtable Backup.
+   *
+   * @param {ModifiableBackupFields} metadata - The fields to be updated.
+   * @param {CallOptions | SetMetadataCallback} [gaxOptionsOrCallback]
+   * @param {SetMetadataCallback} [callback] The callback function.
+   * @param {?error} callback.err An error returned while making this request.
+   * @param
+   * @param {object} callback.apiResponse The full API response.
+   * @return {void | Promise<SetMetadataResponse>}
+   */
+  setMetadata(
+    metadata: ModifiableBackupFields,
+    gaxOptionsOrCallback?: CallOptions | SetMetadataCallback,
+    cb?: SetMetadataCallback
+  ): void | Promise<SetMetadataResponse> {
+    const gaxOpts =
+      typeof gaxOptionsOrCallback === 'object' ? gaxOptionsOrCallback : {};
+    const callback =
+      typeof gaxOptionsOrCallback === 'function' ? gaxOptionsOrCallback : cb!;
+
+    const {expireTime, ...restMetadata} = metadata;
```

Comment:
As of right now the only updatable field is `expireTime`.
In general, do we want to require that "at least one field value should be provided" and throw if none provided?

Reply:
I don't feel too strongly, although I don't think we have a consistent precedent for that. I lean towards letting the API fail if it has an issue, when we don't have any code that requires it be defined (e.g. if we did "expireTime.toStruct()" without making sure if it works first).